### DISCUSSION TITLE
How can I modify the app name and content displayed in the menu bar on macOS?

### DISCUSSION THREAD

**Discussion by luluxiaoyu** [AvaloniaUI/Avalonia#17847] [Q&A] [answered]
![daa608d35a1b409af0f253a22c514a6f](https://github.com/user-attachments/assets/5cb7d1d2-c0b3-43b6-934e-ccb472482674)

**Comment by MikeCodesDotNET** [comment, marked as answer]
The easiest way to do this would be to set the `Name` property in the App.axaml. Something like this would do it: 

```xaml 
<Application xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             x:Class="macOSAppNameDemo.App"
             Name="My App Name"
             xmlns:local="using:macOSAppNameDemo"
             RequestedThemeVariant="Default">
```

This results in the following: 
![CleanShot 2024-12-30 at 07 47 21@2x](https://github.com/user-attachments/assets/a56a337e-54e1-4fb9-9cae-480409425e62)

If you want to then customise the menu, you can use [NativeMenu](https://docs.avaloniaui.net/docs/reference/controls/nativemenu). The following is an example of this (place it inside the <Application> in the `App.axaml`

```xaml 
<NativeMenu.Menu>
     <NativeMenu>
         <NativeMenuItem Header="About My App Name" />
         <NativeMenuItem Header="Check for Updates..." />
     </NativeMenu>
 </NativeMenu.Menu>
 ```
 
![CleanShot 2024-12-30 at 07 51 01@2x](https://github.com/user-attachments/assets/35831bea-c312-42e7-b772-0c04a230c1a9)

**Reply by luluxiaoyu** [reply]
ok, and can the language of the default menu items (such as hidden options) in macOS be modified?
